src/Pelagos/Bundle/AppBundle/Resources/public/js/download.js

```diff
@@ -4,42 +4,104 @@ $(document).ready(function()
 
 function startDownload(id)
 {
-    if ($("#download_splash").length) {
-        return;
-    }
-
     $('<div id="download_splash" />').appendTo("#pelagos-content").load(
-        Routing.generate("pelagos_app_download_default", {"id": id}),
-        function () {
-            initializeDownload(id);
-        }
+        $.getJSON( Routing.generate("pelagos_app_download_default", {"id": id}), function( data ) {
+
```

Comment:
instead of doing all that stuff in .done, you can just do it in your called back. Success is assumed.

Reply:
For the html div to be populated, i need the ajax request to process before the dialog box is created